src/context/DarkModeContext.tsx

```diff
@@ -1,11 +1,15 @@
 import * as React from 'react';
+import { useIsUserDataLoaded } from './UserDataContext/UserDataContext';
 
 // Note: this is separate from DarkModeProvider.tsx because this is imported by Storybook.
 
 export const DarkModeContext = React.createContext<boolean>(null);
 
 export const useDarkMode = (): boolean => {
   const context = React.useContext(DarkModeContext);
+  if (!useIsUserDataLoaded()) {
+    return document.documentElement.classList.contains('dark');
+  }
```

Comment:
Maybe this should be moved into `DarkModeProvider`?

Note that any access to document must be inside a `useEffect`. I think a better approach would be to leave this function alone and modify `DarkModeProvider` instead. DarkModeProvider waits for the user data to load before setting dark mode, which is slow. Instead, before `isLoaded` is set, it should do something similar to `gatsby-ssr.tsx` (ie. try loading from `themeKey` or default to system).

Reply:
Curious; even though the first value is likely still wrong, there's no flicker at all anymore? Do you know what's going on?

Reply:
Ah I see, it's probably because everything on the problems page is only rendered client side 😅